apps/wallet/src/ui/app/redux/slices/originbyte-nfts/NftClient.ts

```diff
@@ -0,0 +1,269 @@
+
+    return response;
+};
+
+export class NftClient {
+    constructor(private provider: JsonRpcProvider) {}
+
+    private fetchObjectIdsForAddress = async <RpcResponse, DataModel>(
+        address: string,
+        parser: SuiObjectParser<RpcResponse, DataModel>
+    ): Promise<string[]> => {
+        const objectsForWallet = await this.provider.getObjectsOwnedByAddress(
+            address
+        );
+        return objectsForWallet
+            .filter((_) => _.type.match(parser.regex))
+            .map((_) => _.objectId);
+    };
+
+    parseObjects = async <RpcResponse, DataModel>(
+        objects: GetObjectDataResponse[],
+        parser: SuiObjectParser<RpcResponse, DataModel>
+    ): Promise<DataModel[]> => {
+        const parsedObjects = objects
+            .filter(isObjectExists)
+            .map((_) => {
+                if (
+                    isSuiObject(_.details) &&
+                    isSuiMoveObject(_.details.data) &&
+                    _.details.data.type.match(parser.regex)
+                ) {
+                    return parser.parser(
+                        _.details.data.fields as RpcResponse,
+                        _.details,
+                        _
+                    );
+                }
+                return undefined;
+            })
+            .filter((_): _ is DataModel => !!_);
+
+        return parsedObjects;
+    };
+
+    fetchAndParseObjectsById = async <RpcResponse, DataModel>(
+        ids: string[],
+        parser: SuiObjectParser<RpcResponse, DataModel>
+    ): Promise<DataModel[]> => {
+        if (ids.length === 0) {
+            return [];
+        }
+        const objects = await this.provider.getObjectBatch(ids);
+        return this.parseObjects(objects, parser);
+    };
+
+    getBagContent = async (bagId: string) => {
+        const bagObjects = await this.provider.getObjectsOwnedByObject(bagId);
+        const objectIds = bagObjects.map((_) => _.objectId);
+        return this.provider.getObjectBatch(objectIds);
+    };
+
+    getAndParseBagContent = async <RpcResponse, DataModel>(
+        bagId: string,
+        parser: SuiObjectParser<RpcResponse, DataModel>
+    ) => {
+        const bagObjects = await this.getBagContent(bagId);
+        return this.parseObjects(bagObjects, parser);
+    };
+
+    getNftsById = async (params: WithIds): Promise<ArtNft[]> => {
+        const nfts = await this.fetchAndParseObjectsById(
+            params.objectIds,
+            ArtNftParser
+        );
+        const bags = await Promise.all(
+            nfts.map(async (_) => {
+                const content = await this.getBagContent(_.bagId);
+                return {
+                    nftId: _.id,
+                    content: parseDomains(content),
+                };
+            })
+        );
+        const bagsByNftId = new Map(bags.map((b) => [b.nftId, b]));
+
+        return nfts.map((nft) => {
+            const fields = bagsByNftId.get(nft.id);
+            return {
```

Comment:
Why not just return the raw `fields` + `nft` and let the consumer extract values that make sense instead of having these limited mapped response structures? (this also applies to the parser return value)